src/dodal/devices/i22/nxsas.py

```diff
@@ -0,0 +1,180 @@
+                    float, rotation_angle[0], units=rotation_angle[1]
+                )
+            else:
+                self.rotation_angle = None
+            if aequetorial_angle is not None:
+                self.aequetorial_angle, _ = soft_signal_r_and_setter(
+                    float, aequetorial_angle[0], units=aequetorial_angle[1]
+                )
+            else:
+                self.aequetorial_angle = None
+            if beam_center_x is not None:
+                self.beam_center_x, _ = soft_signal_r_and_setter(
+                    float, beam_center_x[0], units=beam_center_x[1]
+                )
+            else:
+                self.beam_center_x = None
+            if beam_center_y is not None:
+                self.beam_center_y, _ = soft_signal_r_and_setter(
+                    float, beam_center_y[0], units=beam_center_y[1]
+                )
+            else:
+                self.beam_center_y = None
+            if sensor_material is not None:
+                self.sensor_material, _ = soft_signal_r_and_setter(str, sensor_material)
+            else:
+                self.sensor_material = None
+            if description is not None:
+                self.description, _ = soft_signal_r_and_setter(str, description)
+            else:
+                self.description = None
+            if type is not None:
+                self.type, _ = soft_signal_r_and_setter(str, type)
+            else:
+                self.type = None
+            if serial_number is not None:
+                self.serial_number, _ = soft_signal_r_and_setter(str, serial_number)
+            else:
+                self.serial_number = None
+            if sensor_thickness is not None:
+                self.sensor_thickness, _ = soft_signal_r_and_setter(
+                    float, sensor_thickness[0], units=sensor_thickness[1]
+                )
+            else:
+                self.sensor_thickness = None
+            if threshold_energy is not None:
+                self.threshold_energy, _ = soft_signal_r_and_setter(
+                    float, threshold_energy[0], units=threshold_energy[1]
+                )
+            else:
+                self.threshold_energy = None
+
+
+class NXSasPilatus(PilatusDetector):
+    def __init__(
+        self,
+        prefix: str,
+        directory_provider: DirectoryProvider,
+        drv_suffix,
+        hdf_suffix,
+        metadata_holder: NXSasMetadataHolder,
+        name: str = "",
+    ):
+        super().__init__(
+            prefix,
+            directory_provider,
+            drv_suffix=drv_suffix,
+            hdf_suffix=hdf_suffix,
+            name=name,
+        )
+        self._metadata_holder = metadata_holder
+
+    async def describe_configuration(self) -> Dict[str, DataKey]:
+        return {
+            **await super().describe_configuration(),
+            **await self._metadata_holder.describe_configuration(),
+        }
```

Comment:
pls add a comment what is going on here and why do we need to extra metadata

Reply:
I've added docstrings to the type, is that sufficient?